website/pages/docs/from-1.5/special-variables.mdx

```diff
@@ -0,0 +1,63 @@
+---
+layout: docs
+page_title: Special Variables - HCL Configuration Language
+sidebar_title: Special Variables
+description: |-
+  Special variables provide connection information and basic instance state information.
+  This page covers all existing special variables.
+---
+
+# Build Variables
+
+`@include 'from-1.5/beta-hcl2-note.mdx'`
+
+Build variables will allow you to access, from provisioners and post-processors, connection information and basic instance state information.
+All special build variables are prefixed with `build`:
+
+```hcl
+  provisioner "shell" {
+      environment_vars = ["TESTVAR=${build.PackerRunUUID}"]
+      inline = ["echo $TESTVAR"]
+  }
+```
+
+The valid build variables to request are:
+
+- **ID**: Represents the vm being provisioned. For example, in Amazon it is the instance id; in digitalocean,
+  it is the droplet id; in Vmware, it is the vm name.
+
+- **Host**, **Port**, **User** and **Password**: The host, port, user, and password that Packer uses to access the machine.
+  Useful for using the shell local provisioner to run Ansible or Inspec against the provisioned instance.
+
+- **ConnType**: Type of communicator being used. For example, for SSH communicator this will be "ssh".
+
+- **PackerRunUUID**: Current build's unique id. Can be used to specify build artifacts.
+
+- **PackerHTTPIP**, **PackerHTTPPort**, and **PackerHTTPAddr**: HTTP IP, port, and address of the file server Packer creates to serve items in the "http" dir to the vm. The HTTP address is displayed in the format `IP:PORT`.
+
+- **SSHPublicKey** and **SSHPrivateKey**: The public and private key that Packer uses to connect to the instance.
+  These are unique to the SSH communicator and are unset when using other communicators.
+  **SSHPublicKey** and **SSHPrivateKey** can have escape sequences and special characters so their output should be single quoted to avoid surprises. For example:
```

Comment:
What do you think about making them snake case ?

Reply:
The main argument would be that it keeps sort of a 'packer feeling' throughout the code and a simple map iteration should solve this.

Reply:
Like `SshPublicKey`,  `PackerHttpPort`, `Id`, etc..? 
This is going to make them different from JSON. Is that ok?

Reply:
I think it is, a regexp should be able to fix this somewhat easily

Reply:
> The main argument would be that it keeps sort of a 'packer feeling' throughout the code and a simple map iteration should solve this.

hmm, this sounds like an internal developer optimization that may result in us repeating to users in issues that variable names changed in HCL. I don't see the immediate benefit and think it should probably be held for another PR.

Reply:
Gotcha yes I agree 🙂